src/internal/packager/images/pull.go

```diff
@@ -83,15 +83,15 @@ func (i *ImageConfig) PullAll() ([]ImgInfo, error) {
 				return
 			}
 
-			actualSrc := refInfo.Reference
-			if overrideHost, present := i.RegistryOverrides[refInfo.Host]; present {
-				var err error
-				actualSrc, err = transform.ImageTransformHostWithoutChecksum(overrideHost, refInfo.Reference)
-				if err != nil {
-					metadataImageConcurrency.ErrorChan <- fmt.Errorf("failed to swap override host %s for %s: %w", overrideHost, refInfo.Reference, err)
-					return
+			actualSrc := ""
```

Comment:
We don't know if actualSrc is going to have to change so we should keep it as refInfo.Reference. Otherwise we will also error in the non registry override case.

For this reason we should also remove the `if actualSrc == ""` check

Reply:
what is the non registry override case ?

Reply:
The non registry override case is the default case. The only time the registry is overridden is when the registry on the image matches one of the registries sent through the --registry-overrides flag. 

For example, if I have an image at ghcr.io/defenseunicorns/zarf/agent unless I sent into my package create --registry-override `ghcr.io=mydomain.com` `actualSrc` would have stayed "" and the code would error.

Reply:
```suggestion
			actualSrc := refInfo.Reference
```
actualSrc needs to stay as refInfo.Reference as well otherwise we will pull an empty string when there is no override

Reply:
got it, without the registry override the actualSrc value should be the default reference to the image. When the overriding is happening only then actualSrc should be updated. I have updated the PR. Shall squash and sign the commit.